airbyte-integrations/connectors/source-s3/source_s3/v4/cursor.py

```diff
@@ -16,6 +17,11 @@ class Cursor(DefaultFileBasedCursor):
     _V4_MIGRATION_BUFFER = timedelta(hours=1)
     _V3_MIN_SYNC_DATE_FIELD = "v3_min_sync_date"
 
+    def __init__(self, stream_config: FileBasedStreamConfig, **_: Any):
+        super().__init__(stream_config)
+        self._running_migration = False
```

Comment:
re: am I surprised they were accessed before being set?
A little bit, but in hindsight, it makes sense since `set_initial_state` isn't called on full refresh runs

Reply:
Q: do you know if this was only a problem for full refresh?

either way, the fix looks good

Reply:
It was happening in incremental syncs, too, but during the first run (before there is a state).